File: Java/Advinhe o número/Advinhe.java
import java.util.Random;
import java.util.Scanner;

public class Advinhe {
    public static void main(String[] args) {
        Random random = new Random();
        int numeroAleatorio = random.nextInt(100) + 1; // gera um número aleatório entre 1 e 100
        Scanner scanner = new Scanner(System.in);

        System.out.println("Tente adivinhar o numero entre 1 e 100:");
        int tentativas = 0;

        while (true) {
            int resposta = scanner.nextInt();
            tentativas++;

            if (resposta == numeroAleatorio) {
                System.out.println("Parabens, voce acertou o numero em " + tentativas + " tentativas!");
                break;
            } else if (resposta < numeroAleatorio) {
                System.out.println("O numero e maior! Tente novamente:");
            } else {
                System.out.println("O numero e menor! Tente novamente:");
            }
        }

        scanner.close();
    }
}

File: Python/Advinhe o número/advinhe.py
import random

# gerar um número aleatório entre 1 e 100
numero_aleatorio = random.randint(1, 100)

# inicializar o contador de tentativas
tentativas = 0

# loop para pedir ao usuário para adivinhar o número
while True:
    # pedir ao usuário para digitar um número
    tentativa = int(input("Digite um número entre 1 e 100: "))

    # incrementar o contador de tentativas
    tentativas += 1

    # verificar se o número é igual ao número aleatório
    if tentativa == numero_aleatorio:
        print("Parabéns! Você acertou após", tentativas, "tentativas!")
        break
    # verificar se o número é maior ou menor que o número aleatório
    elif tentativa > numero_aleatorio:
        print("O número é menor. Tente novamente.")
    else:
        print("O número é maior. Tente novamente.")


